connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerTask.java

```diff
@@ -122,7 +127,11 @@ private void doRun() {
             if (stopping.get())
                 return;
 
-            lifecycleListener.onStartup(id);
+            if (targetState.get() == TargetState.PAUSED)
+                statusListener.onPause(id);
+            else
+                statusListener.onStartup(id);
```

Comment:
So we keep calling `onPause` or `onStartup` not just only we the state is changed?